subsys/bluetooth/services/ots/ots_oacp.c

```diff
@@ -298,102 +298,96 @@ static enum bt_gatt_ots_oacp_res_code oacp_proc_validate(
 	}
 };
 
-static enum bt_gatt_ots_oacp_res_code oacp_command_decode(
-	const uint8_t *buf, uint16_t len,
-	struct bt_gatt_ots_oacp_proc *proc)
+static int oacp_command_decode(const uint8_t *buf, uint16_t len,
+			       struct bt_gatt_ots_oacp_proc *proc)
 {
 	struct net_buf_simple net_buf;
 
+	if (len < OACP_PROC_TYPE_SIZE) {
+		return -ENODATA;
+	}
+
 	net_buf_simple_init_with_data(&net_buf, (void *) buf, len);
 
 	proc->type = net_buf_simple_pull_u8(&net_buf);
 	switch (proc->type) {
 #if defined(CONFIG_BT_OTS_OACP_CREATE_SUPPORT)
 	case BT_GATT_OTS_OACP_PROC_CREATE:
```

Comment:
Not directly related to this PR, but I have a feeling you'd get more readable code out of this if you used an operation handler table instead of a long switch statement. In particular, the combination of a switch statement and the #ifdefs for every case makes this difficult to read (at least for me). Anyway, this can be addressed in a separate PR.

Reply:
I will follow up with a PR to clean this up. Thanks for the suggestion.